### ISSUE TITLE
Link color of Site Title clobbered by containing Group link color

### ISSUE THREAD

**Issue by pbking** [WordPress/gutenberg#38860]
### Description

When a group has a link color assigned and in that group a site title has a link color assigned the EDITOR shows the correct color but the VIEW shows the color defined in the GROUP.

### Step-by-step reproduction instructions

Using emptytheme create a page with a group block that contains a site title block.
Assign a link color value to the group block.
Assign a DIFFERENT link color value to the site title block.

Example:
```
<!-- wp:group {"style":{"elements":{"link":{"color":{"text":"var:preset|color|vivid-cyan-blue"}}}}} -->
<div class="wp-block-group has-link-color">
<!-- wp:site-title {"style":{"elements":{"link":{"color":{"text":"var:preset|color|vivid-red"}}}}} /-->
</div>
<!-- /wp:group -->
```

Note that in the EDITOR the site title renders using the color assigned to the SITE TITLE block while in the VIEW the site title renders using the color assigned to the GROUP block.

The difference seems to be caused because the ORDER of the two `a` element selectors is different in the CSS in editor vs view (The editor renders the GROUP's selector first and the SITE TITLE next which causes the value assigned to the SITE TITLE to win, while the view renders the SITE TITLE's selector first and the GROUP's second causing the GROUP to win.) 

See the screenshots for more details.

### Screenshots, screen recording, code snippet

Editor:

![image](https://user-images.githubusercontent.com/146530/154314957-b7f1a940-1af4-4e29-9a12-77edf84fefad.png)

View:
![image](https://user-images.githubusercontent.com/146530/154315616-7af36499-2a5e-480b-8dd1-7771365817a6.png)

### Environment info

Tested in Gutenberg 12.5 as well as /trunk (currently 12.6.0-RC1 I believe)
WP 5.9

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by skorasaurus** [comment]
similar to https://github.com/WordPress/gutenberg/issues/37071 ?

**Comment by Smit2808** [comment]
Hi @pbking 

I checked this issue and now it is working fine for the page, but if we edit the FSE and then change the site title link color and publish it, in the editor it will work but on the front side, the color is not changing.

My environment:
-> WP V5.9.2
-> GB V12.8.1
-> Browser chrome
-> MAC OS

Thank you.

**Comment by pbking** [comment]
Closing as fixed by #37072
